parsers/cpreprocessor.c

```diff
@@ -1565,15 +1565,15 @@ static void saveMacro(const char * macro)
 		return;
 	}
 
-	if(!(isalpha(*c) || (*c == '_')))
+	if(!(isalpha(*c) || (*c == '_' || (*c == '$') )))
```

Comment:
Could you fix the parenthesis so (*c == '$') is at the same nesting level of (*c == '_') ?